colcon_core/package_identification/python.py

```diff
@@ -2,12 +2,26 @@
 # Licensed under the Apache License, Version 2.0
 
 import re
+import sys
 
 from colcon_core.package_identification import logger
 from colcon_core.package_identification \
     import PackageIdentificationExtensionPoint
 from colcon_core.plugin_system import satisfies_version
-from setuptools.config import read_configuration
+try:
+    from setuptools.config import read_configuration
+except ImportError:
+    from pkg_resources import get_distribution
+    from pkg_resources import parse_version
+    setuptools_version = get_distribution('setuptools').version
+    minimum_version = '30.3.0'
+    if parse_version(setuptools_version) < parse_version(minimum_version):
+        sys.exit(
```

Comment:
`sys.exit()` is kind of harsh for an `ImportError`, but it does make the message easy to read when trying to bootstrap without installing setuptools.

```
$ ./src/colcon-core/bin/colcon
'setuptools' needs to be at least version 30.3.0, if a newer version is not available from the package manager use 'pip3 install -U setuptools' to update to the latest version
```

 I'm ok with it, but would prefer a raised `ImportError` instead.

Reply:
I modified the patch to extend the raised `ModuleNotFound` exception with additional information where applicable: https://github.com/colcon/colcon-core/pull/91/commits/444e88edb9b2245d54388b37b751a1f821a6959a#diff-9fc4dc6d196841f3b396afba788b9246R18

The "downside" is that the build will continue, likely falling back to the `python_setup_py` extension which fails for every package using a `setup.cfg` file rather than declaring the package name in the `setup.py` file.

Reply:
`ModuleNotFoundError` is actually only available as of Pytohn 3.6. Therefore I updated the patch to catch the more generic exception `ImportError`.